packages/api/cms-api/src/generator/utils/generate-required-permission-decorators.ts

```diff
@@ -0,0 +1,45 @@
+import { CrudGeneratorOptions } from "../crud-generator.decorator";
+
+export function generateRequiredPermissionDecorators({
+    generatorOptions,
+    hasScopeProp,
+}: {
+    generatorOptions: CrudGeneratorOptions;
+    hasScopeProp: boolean;
+}) {
+    const ret: {
+        resolverDecorator?: string;
+        listDecorator?: string;
+        singleDecorator?: string;
+        createDecorator?: string;
+        updateDecorator?: string;
+        deleteDecorator?: string;
+    } = {};
+    if (generatorOptions.requiredPermission) {
+        const requiredPermission = generatorOptions.requiredPermission;
+        if (typeof requiredPermission === "string" || Array.isArray(requiredPermission)) {
+            ret.resolverDecorator = `@RequiredPermission(${JSON.stringify(
+                Array.isArray(requiredPermission) ? requiredPermission : [requiredPermission],
```

Comment:
if requiredPermission is a string (and not an array) the decorator should also have a string (and not an array)

doesn't make a difference but if you use the generated code as scaffolding it's nicer to omit this unnecessary array

Reply:
currently RequiredPermission-decorator is defined to only support array: https://github.com/vivid-planet/comet/blob/main/packages/api/cms-api/src/user-permissions/decorators/required-permission.decorator.ts#L27

so if a generator-config permission-config defined only a string, the resulting code wasn't correct.

Reply:
ouch, here https://github.com/vivid-planet/comet/blob/main/packages/api/cms-api/src/user-permissions/decorators/required-permission.decorator.ts#L23 it is allowed

Reply:
I added the change as this is a bugfix required for this pull-request.

Reply:
moved it into another pull-request: https://github.com/vivid-planet/comet/pull/1673

Reply:
single permission is now possible in main